apiserver/facades/client/application/application.go

```diff
@@ -2465,23 +2470,35 @@ func (api *APIBase) unitResultForUnit(ctx context.Context, unitName coreunit.Nam
 	if errors.Is(err, applicationerrors.UnitNotFound) {
 		return nil, errors.NotFoundf("unit %s", unitName)
 	} else if errors.Is(err, applicationerrors.UnitMachineNotAssigned) {
-		unit, err := api.backend.Unit(unitName.String())
-		if err != nil {
+		providerID, err := api.applicationService.GetUnitProviderID(ctx, unitName)
```

Comment:
Could be better to have a method that returns a struct representing the k8s pod? Feels weird to have 2 separate db hits to get a single column each from the same table.

Reply:
Not quite correct, each of the three vars here come from different join tables, so if we pushed them down we would still require 3 queries.

Happy to push them down nonetheless, `GetUnitContainerInfo` would be a valid abstraction, but I don't think it''s too important imo

Reply:
Sorry I wasn't clear. When I say "2 separate db hits" I was referring to the providerID and the port, which both come from the `k8s_pod` table. Hence having a containerInfo-like request.

Reply:
> Happy to push them down nonetheless, GetUnitK8sPodInfo would be a valid abstraction, if you would like

Do this.

Reply:
Done!